src/ui/session/content/message_row/bubble.rs

```diff
@@ -225,18 +230,22 @@ impl MessageBubble {
             binding.unwatch();
         }
 
-        let sender_binding = model::Chat::this_expression("title").bind(
-            &*imp.sender_label,
-            "label",
-            Some(&sponsored_message.sponsor_chat_()),
-        );
-        imp.sender_binding.replace(Some(sender_binding));
+        imp.sender_label
+            .set_label(&sponsored_message.sponsor_label());
+
+        let label_hash = self.hash_label(sponsored_message.sponsor_label());
 
-        self.update_sender_color(Some(sponsored_message.sponsor_chat_().id()));
+        self.update_sender_color(Some(label_hash));
 
         imp.sender_label.set_visible(true);
     }
 
+    fn hash_label(&self, label: String) -> i64 {
```

Comment:
`&self` is not needed here. Also, as you don't intend to own the String `label` you should use a `&str` to avoid copying. Then you should move this function out of the `impl MessageBubble` block.

```suggestion
    fn hash_label(label: &str) -> i64 {
```